packages/workbox-core/_private/DBWrapper.mjs

```diff
@@ -0,0 +1,351 @@
+   * value the callback chooses. The callback holds all the transaction logic
+   * and is invoked with three arguments:
+   *   1. An object mapping object store names to IDBObjectStore values.
+   *   2. A `done` function, that's used to resolve the promise when
+   *      when the transaction is done.
+   *   3. An `abort` function that can be called to abort the transaction
+   *      at any time.
+   *
+   * @param {Array<string>} storeNames An array of object store names
+   *     involved in the transaction.
+   * @param {string} type Can be `readonly` or `readwrite`.
+   * @param {function(Object, function(), function(*)):?IDBRequest} callback
+   * @return {*} The result of the transaction ran by the callback.
+   */
+  async transaction(storeNames, type, callback) {
+    await this.open();
+    const result = await new Promise((resolve, reject) => {
+      const txn = this._db.transaction(storeNames, type);
+      const done = (value) => resolve(value);
+      const abort = () => {
+        reject(new Error('The transaction was manually aborted'));
+        txn.abort();
+      };
+      txn.onerror = (evt) => reject(evt.target.error);
+      txn.onabort = (evt) => reject(evt.target.error);
+      txn.oncomplete = () => resolve();
+
+      const stores = {};
+      for (const storeName of storeNames) {
+        stores[storeName] = txn.objectStore(storeName);
+      }
+      callback(stores, done, abort);
+    });
+    return result;
+  }
+
+  /**
+   * Delegates async to a native IDBObjectStore method.
+   *
+   * @param {string} method The method name.
+   * @param {string} storeName The object store name.
+   * @param {string} type Can be `readonly` or `readwrite`.
+   * @param {...*} args The list of args to pass to the native method.
+   * @return {*} The result of the transaction.
+   */
+  async _call(method, storeName, type, ...args) {
+    await this.open();
+    const callback = (stores, done) => {
+      stores[storeName][method](...args).onsuccess = (evt) => {
+        done(evt.target.result);
+      };
+    };
+
+    return await this.transaction([storeName], type, callback);
+  }
+
+  /**
+   * The default onversionchange handler, which closes the database so other
+   * connections can open without being blocked.
+   *
+   * @param {Event} evt
+   */
+  _onversionchange(evt) {
+    this.close();
+  }
+
+  /**
+   * Closes the connection opened by `DBWrapper.open()`. Generally this method
+   * doesn't need to be called since:
+   *   1. It's usually better to keep a connection open since opening
+   *      a new connection is somewhat slow.
+   *   2. Connections are automatically closed when the reference is
+   *      garbage collected.
+   * The primary use case for needing to close a connection is when another
+   * reference (typically in another tab) needs to upgrade it and would be
+   * blocked by the current, open connection.
+   */
+  close() {
+    if (this._db) this._db.close();
```

Comment:
I guess it's not something linting picks up, but I'm under the impression that we try to avoid single-line `if()`s.

Reply:
This is allowed in Google style, see [section 4.1.1](https://google.github.io/styleguide/jsguide.html#formatting-braces-all).

I prefer the single-line format for things that are conceptually simple like this. Personally I find that when these lines take up too much room in a function, the logic is harder to follow.

I'm happy to change if folks all disagree, but I wanted to point out that it's not against Google JS style.

Reply:
Okay, thanks for clarifying. I'm cool if our style guide is cool (though I don't know that I'm going to start doing it).